File: mocks/oppgave-mock/src/main/java/no/nav/pensjon/vtp/mocks/oppgave/rest/infrastruktur/validering/OneOfValidator.java
package no.nav.pensjon.vtp.mocks.oppgave.rest.infrastruktur.validering;

import javax.validation.ConstraintValidator;
import javax.validation.ConstraintValidatorContext;
import java.util.Arrays;

@SuppressWarnings("WeakerAccess")
public class OneOfValidator implements ConstraintValidator<OneOf, String> {
    private String[] legalValues;

    @Override
    public void initialize(OneOf oneOf) {
        this.legalValues = oneOf.legalValues();
    }

    @Override
    public boolean isValid(String oneOf, ConstraintValidatorContext constraintValidatorContext) {
        if (isBlank(oneOf)) {
            return true;
        }

        return Arrays.stream(legalValues).anyMatch(legalEntry -> legalEntry.equals(oneOf));
    }

    public static boolean isBlank(final String string) {
        return string.chars().allMatch(Character::isWhitespace);
    }
}

File: mocks/oppgave-mock/src/main/java/no/nav/pensjon/vtp/mocks/oppgave/rest/infrastruktur/validering/OneOfValidator.kt
package no.nav.pensjon.vtp.mocks.oppgave.rest.infrastruktur.validering

import java.util.*
import javax.validation.ConstraintValidator
import javax.validation.ConstraintValidatorContext

class OneOfValidator : ConstraintValidator<OneOf, String> {
    private lateinit var legalValues: Array<String>

    override fun initialize(oneOf: OneOf) {
        legalValues = oneOf.legalValues
    }

    override fun isValid(oneOf: String, constraintValidatorContext: ConstraintValidatorContext): Boolean {
        return if (isBlank(oneOf)) {
            true
        } else Arrays.stream(legalValues).anyMatch { legalEntry: String -> legalEntry == oneOf }
    }

    companion object {
        fun isBlank(string: String): Boolean {
            return string.chars().allMatch { codePoint: Int -> Character.isWhitespace(codePoint) }
        }
    }
}
